site/src/pages/CreateTemplatePage/VariableInput.tsx

```diff
@@ -0,0 +1,137 @@
+import FormControlLabel from "@material-ui/core/FormControlLabel"
+import Radio from "@material-ui/core/Radio"
+import RadioGroup from "@material-ui/core/RadioGroup"
+import { makeStyles } from "@material-ui/core/styles"
+import TextField from "@material-ui/core/TextField"
+import { Stack } from "components/Stack/Stack"
+import { FC } from "react"
+import { TemplateVersionVariable } from "../../api/typesGenerated"
+
+const isBoolean = (variable: TemplateVersionVariable) => {
+  return variable.type === "bool"
+}
+
+const VariableLabel: React.FC<{ variable: TemplateVersionVariable }> = ({
+  variable,
+}) => {
+  const styles = useStyles()
+
+  return (
+    <label htmlFor={variable.name}>
+      <span className={styles.labelName}>
+        var.{variable.name}
+        {!variable.required && " (optional)"}
+      </span>
+      <span className={styles.labelDescription}>{variable.description}</span>
+    </label>
+  )
+}
+
+export interface VariableInputProps {
+  disabled?: boolean
+  variable: TemplateVersionVariable
+  onChange: (value: string) => void
+  defaultValue?: string
+}
+
+export const VariableInput: FC<VariableInputProps> = ({
```

Comment:
Shouldn't this component be placed in `components`?

Reply:
This is a good question. We should try to keep code close to where it is used, if this component starts to be used in other pages like the EditTemplatePage, we should move it to the components folder.